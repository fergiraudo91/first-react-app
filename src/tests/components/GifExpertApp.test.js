import React from 'react';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../../GifExpertApp';
import { shallow } from 'enzyme';

describe('Pruebas en GifExpertApp', () => {
    const wrapper = shallow(<GifExpertApp />);
   test('Debe hacer match con el snatshop', () => {
       expect(wrapper).toMatchSnapshot();
   });

   test('Debe mostrar una lista de categorias', () => {
       const categories = ['shingeki no kyojin', 'Akame Ga Kill'];
       const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
       expect(wrapper).toMatchSnapshot();
       expect(wrapper.find('GifGrid').length).toBe(categories.length);
   })
   
    
});
