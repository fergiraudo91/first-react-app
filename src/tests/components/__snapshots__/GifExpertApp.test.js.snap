// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe hacer match con el snatshop 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en GifExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="shingeki no kyojin"
      key="shingeki no kyojin"
    />
    <GifGrid
      category="Akame Ga Kill"
      key="Akame Ga Kill"
    />
  </ol>
</Fragment>
`;
